.github/workflows/deploy_whole_doc.yml

```diff
@@ -0,0 +1,129 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Deploy the whole documentation
+
+on:
+  push:
+    branches: [gha-ci, master, develop]
+
+jobs:
+  prepare-matrix:
+    name: Prepare matrix by getting the repo names from repositories.yml
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install yq
+        run: |
+          sudo apt-get update
+          sudo apt-get install python python3-pip
+          pip3 install yq
+        shell: bash
+      - name: Get repo names
+        id: set-matrix
+        run: |
+          export REPOS=$(cat .repos/repositories.yml | yq -c '[.[]]')
+          echo "::set-output name=repo::$REPOS"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.repo }}
+
+  deploy_framework:
+    name: Deploy framework
+    runs-on: ubuntu-latest
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_DEFAULT_REGION: us-west-1
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+      - name: Install dependencies
+        run: npm ci
+        shell: bash
+      - name: Build framework
+        run: npm run build
+        shell: bash
+      - name: Upload framework to S3 (PROD)
+        if: ${{ github.ref == 'refs/heads/master' }}
+        run: aws s3 sync src/.vuepress/dist s3://docs.kuzzle.io
+        shell: bash
+      - name: Upload framework to S3 (NEXT)
+        if: ${{ github.ref != 'refs/heads/master' }}
+        run: aws s3 sync src/.vuepress/dist s3://docs-next.kuzzle.io
+        shell: bash
+      - name: Invalidate Cloudfront cache (PROD)
+        if: ${{ github.ref == 'refs/heads/master' }}
+        shell: bash
+        run: aws cloudfront create-invalidation --distribution-id E3D6RP0POLCJMM --paths '/*'
+      - name: Invalidate Cloudfront cache (NEXT)
+        if: ${{ github.ref != 'refs/heads/master' }}
+        shell: bash
+        run: aws cloudfront create-invalidation --distribution-id E2ZCCEK9GRB49U --paths '/*'
```

Comment:
This whole section could be done in a dedicated action

Reply:
What would be the benefits?

Reply:
As it you could avoid multiple `if` conditions and improve readability

Example of this workflow using a dedicated GHA:
This:
```yaml
      - name: Install dependencies
        run: npm ci
        shell: bash
      - name: Build framework
        run: npm run build
        shell: bash
      - name: Upload framework to S3 (PROD)
        if: ${{ github.ref == 'refs/heads/master' }}
        run: aws s3 sync src/.vuepress/dist s3://docs.kuzzle.io
        shell: bash
      - name: Upload framework to S3 (NEXT)
        if: ${{ github.ref != 'refs/heads/master' }}
        run: aws s3 sync src/.vuepress/dist s3://docs-next.kuzzle.io
        shell: bash
      - name: Invalidate Cloudfront cache (PROD)
        if: ${{ github.ref == 'refs/heads/master' }}
        shell: bash
        run: aws cloudfront create-invalidation --distribution-id E3D6RP0POLCJMM --paths '/*'
      - name: Invalidate Cloudfront cache (NEXT)
        if: ${{ github.ref != 'refs/heads/master' }}
        shell: bash
        run: aws cloudfront create-invalidation --distribution-id E2ZCCEK9GRB49U --paths '/*'
```
Could be replaced by this in the workflow file:
```yaml
      - name: Deploy framework (PROD)
         if: ${{ github.ref == 'refs/heads/master' }}
         use: myAction
         with:
           DISTRIBUTION_ID: <PROD_DISTRIBUTION_ID>
           S3_BUCKET: <PROD_BUCKET>
           SRC_FILES: src/.vuepress/dist
      - name: Deploy framework (NEXT)
         if: ${{ github.ref != 'refs/heads/master' }}
         use: myAction
         with:
           DISTRIBUTION_ID: <NEXT_DISTRIBUTION_ID>
           S3_BUCKET: <NEXT_BUCKET>
           SRC_FILES: src/.vuepress/dist
```

And GHA command could look like this:
```yaml
name: Deploy documentation artifacts
description: Deploy documentation artifacts 

runs:
  using: "composite"
  steps:
      - name: Install dependencies
        run: npm ci
        shell: bash
      - name: Build framework
        run: npm run build
        shell: bash
      - name: Upload to S3
        run: aws s3 sync ${{ inputs.SRC_FILES }} ${{ inputs.S3_BUCKET }}
      - name: Invalidate Cloudfront cache
        shell: bash
        run: aws cloudfront create-invalidation --distribution-id ${{ inputs.CLOUDFRONT_ID }} --paths '/*'
```